File: 2026/Company/Company.Web/Modules/ServerTypes/Sales/MeasuresColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { MeasuresRow } from "./MeasuresRow";

export interface MeasuresColumns {
    MeasureId: Column<MeasuresRow>;
    MeasureName: Column<MeasuresRow>;
}

export class MeasuresColumns extends ColumnsBase<MeasuresRow> {
    static readonly columnsKey = 'Sales.Measures';
    static readonly Fields = fieldsProxy<MeasuresColumns>();
}

File: 2026/Company/Company.Web/Modules/Sales/Measures/MeasuresColumns.cs
namespace Company.Sales.Columns;

[ColumnsScript("Sales.Measures")]
[BasedOnRow(typeof(MeasuresRow), CheckNames = true)]
public class MeasuresColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int MeasureId { get; set; }
    [EditLink]
    public string MeasureName { get; set; }
}